contracts/yieldStrategy/gmx/GMXBatchingManager.sol

```diff
@@ -118,19 +126,20 @@ contract GMXBatchingManager is IGMXBatchingManager, OwnableUpgradeable, Pausable
             glpStaked = _stakeGlp(token, amount);
 
             //Update round and glp balance for current round
-            userDeposit.round = currentRound;
+            userDeposit.round = state.currentRound;
             userDeposit.glpBalance = userGlpBalance + glpStaked.toUint128();
-            roundGlpBalance += glpStaked.toUint128();
+            state.roundGlpBalance += glpStaked.toUint128();
         }
-        emit DepositToken(currentRound, token, receiver, amount, glpStaked);
+        emit DepositToken(state.currentRound, token, receiver, amount, glpStaked);
     }
 
-    function executeBatchDeposit() external {
+    function executeBatchDeposit(IERC4626 gmxVault) external {
```

Comment:
If we execute for one vault, it would unpause the deposits. Should we handle this?